snap-core/src/main/java/org/esa/snap/core/util/GeoUtils.java

```diff
@@ -0,0 +1,639 @@
+    }
+
+    private static int getFirstValid_Y(GeoCoding gc, double insetDistance, int xStart, int yStart, int xEnd, int yEnd) {
+        for (int y = yStart; y < yEnd; y++) {
+            for (int x = xStart; x < xEnd; x++) {
+                final PixelPos pixelPos = new PixelPos(x + insetDistance, y + insetDistance);
+                final GeoPos geoPos = gc.getGeoPos(pixelPos, null);
+                if (geoPos.isValid()) {
+                    return y;
+                }
+            }
+        }
+        return yEnd;
+    }
+
+    private static int getLastValid_Y(GeoCoding gc, double insetDistance, int xStart, int yStart, int xEnd, int yEnd) {
+        for (int y = yEnd; y >= yStart; y--) {
+            for (int x = xStart; x < xEnd; x++) {
+                final PixelPos pixelPos = new PixelPos(x + insetDistance, y + insetDistance);
+                final GeoPos geoPos = gc.getGeoPos(pixelPos, null);
+                if (geoPos.isValid()) {
+                    return y;
+                }
+            }
+        }
+        return yStart;
+    }
+
+    private static int getFirstValid_X(GeoCoding gc, double insetDistance, int xStart, int yStart, int xEnd, int yEnd) {
+        for (int x = xStart; x < xEnd; x++) {
+            for (int y = yStart; y < yEnd; y++) {
+                final PixelPos pixelPos = new PixelPos(x + insetDistance, y + insetDistance);
+                final GeoPos geoPos = gc.getGeoPos(pixelPos, null);
+                if (geoPos.isValid()) {
+                    return x;
+                }
+            }
+        }
+
+        return xEnd;
+    }
+
+    private static int getLastValid_X(GeoCoding gc, double insetDistance, int xStart, int yStart, int xEnd, int yEnd) {
+        for (int x = xEnd; x >= xStart; x--) {
+            for (int y = yStart; y < yEnd; y++) {
+                final PixelPos pixelPos = new PixelPos(x + insetDistance, y + insetDistance);
+                final GeoPos geoPos = gc.getGeoPos(pixelPos, null);
+                if (geoPos.isValid()) {
+                    return x;
+                }
+            }
+        }
+
+        return xEnd;
+    }
+
+    /**
+     * Creates a rectangular boundary expressed in pixel positions for the given source rectangle. If the source
+     * {@code rect} is 100 x 50 pixels and {@code step} is 10 the returned array will countain exactly 2 * 10
+     * + 2 * (5 - 2) = 26 pixel positions.
+     * <p>
+     * This method is used for an intermediate step when determining a product boundary expressed in geographical
+     * co-ordinates.
+     * <p>
+     *
+     * @param rect           the source rectangle
+     * @param step           the mean distance from one pixel position to the other in the returned array
+     * @param usePixelCenter {@code true} if the pixel center should be used
+     * @return the rectangular boundary
+     */
+    static PixelPos[] createRectBoundary(final Rectangle rect, int step, final boolean usePixelCenter) {
```

Comment:
Shouldn't this be named createPixelBoundary(...)?

Reply:
same, see above